images/sds-replicated-volume-controller/src/pkg/controller/linstor_resources_watcher.go

```diff
@@ -206,30 +219,103 @@ func ReconcileParams(
 	log.Info("[ReconcileParams] ends work")
 }
 
-func ReconcileTieBreaker(
+func ReconcilePVReplicas(
 	ctx context.Context,
 	log logger.Logger,
+	cl client.Client,
 	lc *lapi.Client,
 	rds map[string]lapi.ResourceDefinitionWithVolumeDefinition,
 	rgs map[string]lapi.ResourceGroup,
+	res map[string][]lapi.Resource,
+	pvs []core.PersistentVolume,
 ) {
-	log.Info("[ReconcileTieBreaker] starts work")
+	log.Info("[ReconcilePVReplicas] starts work")
+
+	for _, pv := range pvs {
+		if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == PVCSIDriver {
+			RGName := rds[pv.Name].ResourceGroupName
+			rg := rgs[RGName]
+			log.Debug(fmt.Sprintf("[ReconcilePVReplicas] PV: %s, RG: %s", pv.Name, rg.Name))
+
+			replicasErrLevel := checkPVMinReplicasCount(ctx, log, lc, rg, res[pv.Name])
+
+			if pv.Labels == nil {
+				pv.Labels = make(map[string]string)
```

Comment:
I'll choose this line as an example. Review point is about problem between `ReconcileParams` and `ReconcilePVReplicas`.

`pv` variable holds a local value, which was copied from the original slice `pvs`. It's not a pointer, and you're not accessing the original array element by index, so any modifications to it will affect only this local copy. I'll use term _value semantics_ to refer to this effect.

_Value semantics_  is fine for `ReconcilePVReplicas` on it's own, and it was fine for `ReconcileParams`, but after you've extended the scope of the state (moved `pvs` to the outer scope), it became a problem, since any change to labels won't affect the `pvs` collection. It will lead to second method `ReconcilePVReplicas` to overwrite the labels set by the `ReconcileParams`.

This should be fixed by using _reference semantics_.